lib/SftpClient.js

```diff
@@ -58,3 +54,45 @@ class SftpClient {
 }
 
 module.exports = SftpClient
+
+async function createReadStream (client, path, session) {
+  var createReadStreamCmd = function (resolve, reject, conn) {
+    return function (err, sftp) {
+      if (err) { return reject(err) }
+      sftp.stat(path, function (err, stat) {
+        if (err) { return reject(err) }
+        var bytes = stat.size
+        if (bytes > 0) {
+          bytes -= 1
+        }
+        try {
+          var stream = sftp.createReadStream(path, { start: 0, end: bytes })
+        } catch (err) {
+          return reject(err)
+        }
+        stream.on('close', function () {
+          // if there is no session we need to clean the connection
+          if (!session) {
+            conn.end()
+            conn.destroy()
+          }
+        })
+        stream.on('error', function () {
+          if (!session) {
+            conn.end()
+            conn.destroy()
+          }
+        })
+
+        // Replaced this
+        // stream.on('readable', function () {
+        //   resolve(stream)
+        // })
+
+        // By this
+        resolve(stream)
```

Comment:
Here is the change I had to make.

@bergos Do you see a potential issue with this change? And do you think I should fork sftp-promises instead of copying the function over?

Reply:
Waiting for `readable` doesn't make sense, so your change looks better to me than the code from the library.

I would propose to keep the code temporary in our repo, try to make a PR (I know the last change is very old, but maybe that small change gets merged), if that doesn't work we can make a fork.